modules/fedimint-ln-client/src/pay.rs

```diff
@@ -404,12 +401,10 @@ impl LightningPayRefundable {
         global_context: DynGlobalClientContext,
     ) {
         // TODO: Remove polling
+        let client = global_context.client();
+        let (_, instance) = client.get_first_module::<LightningClientModule>(&KIND);
         loop {
-            let contract = global_context
-                .api()
-                .with_module(LEGACY_HARDCODED_INSTANCE_ID_LN)
-                .get_outgoing_contract(contract_id)
-                .await;
+            let contract = instance.api.get_outgoing_contract(contract_id).await;
```

Comment:
1. The state transition belongs to a *particular instance*, we cannot just get the first instance here
2. There is no need to get a whole client to get an API. The global context already has both the API struct as well as the instance id of the module the state machine belongs to.